From 2d66dba387f1240ad489b41d9b7504ab2a56dbf9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Elan=20Ruusam=C3=A4e?= <glen@delfi.ee>
Date: Wed, 9 Sep 2015 16:22:56 +0300
Subject: [PATCH] add .xz support for h2ext to be able to build from tar.xz
 templates

magic taken from
https://github.com/file/file/blob/FILE5_24/magic/Magdir/compress#L202-L205

however due limited support of magic format in util-vserver had to hack
some things:

- does not support 'ustring', used 'string'
- does not support \xXX hex escape, used octal
- unable parse octal properly \375 + 7 would be parsed as \3757, so encoded '7' as \67

$ head -c6 < pld32.tar.xz|od -cx
0000000 375   7   z   X   Z  \0
           37fd    587a    005a
---
 distrib/misc/h2ext.desc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/distrib/misc/h2ext.desc b/distrib/misc/h2ext.desc
index a234339..9441bd2 100644
--- a/distrib/misc/h2ext.desc
+++ b/distrib/misc/h2ext.desc
@@ -17,3 +17,5 @@
 # dump/restore old-fs big and little endian
 24	belong		60011			restore -rf -			0
 24	lelong		60011			restore -rf -			0
+# xz
+0	string		\375\67zXZ\0		xz -d -c			1
